test/v4/controllers/createUkPropertyPeriodSummary/def2/Def2_CreateUkPropertyPeriodSummaryValidatorSpec.scala

```diff
@@ -0,0 +1,530 @@
+    Def1_CreateUkPropertyPeriodSummaryRequestBody("2024-04-06", "2024-07-05", None, Some(parsedUkNonFhlPropertyMinimalDef1))
+
+  private val parsedBody =
+    Def2_CreateUkPropertyPeriodSummaryRequestBody("2024-04-06", "2024-07-05", Some(parsedUkFhlProperty), Some(parsedUkNonFhlProperty))
+
+  private val parsedBodyConsolidated =
+    parsedBody.copy(ukFhlProperty = Some(parsedUkFhlPropertyConsolidated), ukNonFhlProperty = Some(parsedUkNonFhlPropertyConsolidated))
+
+  private val parsedBodyMinimalFhl    = parsedBody.copy(ukFhlProperty = Some(parsedUkFhlPropertyMinimal), ukNonFhlProperty = None)
+  private val parsedBodyMinimalNonFhl = parsedBody.copy(ukFhlProperty = None, ukNonFhlProperty = Some(parsedUkNonFhlPropertyMinimal))
+
+  private val validatorFactory = new CreateUkPropertyPeriodSummaryValidatorFactory(mockAppConfig)
+
+  private def validator(nino: String, taxYear: String, businessId: String, body: JsValue) =
+    validatorFactory.validator(nino, businessId, taxYear, body)
+
+  private def setupMocks(): Unit = MockedAppConfig.minimumTaxV2Uk.returns(TaxYear.starting(2022)).anyNumberOfTimes()
+
+  "validator" should {
+    "return the parsed domain object" when {
+      "passed a valid request" in {
+        setupMocks()
+        val result: Either[ErrorWrapper, CreateUkPropertyPeriodSummaryRequestData] =
+          validator(validNino, validTaxYear, validBusinessId, validBody).validateAndWrapResult()
+
+        result shouldBe Right(Def2_CreateUkPropertyPeriodSummaryRequestData(parsedNino, parsedBusinessId, parsedTaxYear, parsedBody))
+      }
+
+      "passed a valid request with a consolidated body for 2024-25" in {
+        setupMocks()
+        val result: Either[ErrorWrapper, CreateUkPropertyPeriodSummaryRequestData] =
+          validator(validNino, validTaxYear, validBusinessId, validBodyConsolidatedWithExtraFields).validateAndWrapResult()
+
+        result shouldBe Right(Def2_CreateUkPropertyPeriodSummaryRequestData(parsedNino, parsedBusinessId, parsedTaxYear, parsedBodyConsolidated))
+      }
+
+      "passed a valid request with a minimal fhl body" in {
+        setupMocks()
+        val result: Either[ErrorWrapper, CreateUkPropertyPeriodSummaryRequestData] =
+          validator(validNino, validTaxYear, validBusinessId, validMinimalFhlBody).validateAndWrapResult()
+
+        result shouldBe Right(Def2_CreateUkPropertyPeriodSummaryRequestData(parsedNino, parsedBusinessId, parsedTaxYear, parsedBodyMinimalFhl))
+      }
+
+      "passed a valid request with a minimal non-fhl body" in {
+        setupMocks()
+        val result: Either[ErrorWrapper, CreateUkPropertyPeriodSummaryRequestData] =
+          validator(validNino, validTaxYear, validBusinessId, validMinimalNonFhlBody).validateAndWrapResult()
+
+        result shouldBe Right(Def2_CreateUkPropertyPeriodSummaryRequestData(parsedNino, parsedBusinessId, parsedTaxYear, parsedBodyMinimalNonFhl))
+      }
+
+      "passed the minimum supported taxYear" in {
```

Comment:
I am not sure if this test should be here, considering this Def2 Validator will never be given any tax year below 24-25